plugins/lead-resources/src/components/MyLeads.svelte

```diff
@@ -0,0 +1,156 @@
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { EmployeeAccount } from '@hcengineering/contact'
+  import { AttachedDoc, Class, DocumentQuery, getCurrentAccount, Ref } from '@hcengineering/core'
+  import { createQuery, getClient } from '@hcengineering/presentation'
+  import task, { Task } from '@hcengineering/task'
+  import { IModeSelector, Label, resolvedLocationStore, SearchEdit, ModeSelector, Loading } from '@hcengineering/ui'
+  import {
+    FilterButton,
+    getViewOptions,
+    makeViewletKey,
+    TableBrowser,
+    viewOptionStore
+  } from '@hcengineering/view-resources'
+  import { IntlString } from '@hcengineering/platform'
+  import ViewletSettingButton from '@hcengineering/view-resources/src/components/ViewletSettingButton.svelte'
+  import view, { Viewlet, ViewletPreference } from '@hcengineering/view'
+  import { onDestroy } from 'svelte'
+  import FilterBar from '@hcengineering/view-resources/src/components/filter/FilterBar.svelte'
+  import lead from '../plugin'
+  import { Lead } from '@hcengineering/lead'
+
+  export let _class: Ref<Class<Lead>> = lead.class.Lead
+  export let labelTasks = lead.string.MyLeads
+
+  let search = ''
+  const currentUser = getCurrentAccount() as EmployeeAccount
+  const assigned = { assignee: currentUser.employee }
+  const created = { createdBy: currentUser._id }
+  let subscribed = { _id: { $in: [] as Ref<Task>[] } }
+
+  $: baseQuery = updateBaseQuery(mode, { assigned, created, subscribed })
+  function updateBaseQuery (mode: string, queries: { [key: string]: DocumentQuery<Lead> }) {
+    return { ...queries[mode] }
+  }
+  let searchQuery: DocumentQuery<Lead> = { ...baseQuery }
+  function updateSearchQuery (search: string): void {
+    searchQuery = search === '' ? { ...baseQuery } : { ...baseQuery, $search: search }
+  }
+  $: if (baseQuery) updateSearchQuery(search)
+  $: resultQuery = { ...searchQuery }
+
+  const subscribedQuery = createQuery()
+  function getSubscribed () {
+    subscribedQuery.query(
+      _class,
+      { 'notification:mixin:Collaborators.collaborators': getCurrentAccount()._id },
+      (result) => {
+        const newSub = result.map((p) => p._id as Ref<AttachedDoc> as Ref<Lead>)
+        const curSub = subscribed._id.$in
+        if (curSub.length !== newSub.length || curSub.some((id, i) => newSub[i] !== id)) {
+          subscribed = { _id: { $in: newSub } }
+        }
+      },
+      { sort: { _id: 1 } }
+    )
+  }
+  $: if (mode === 'subscribed') getSubscribed()
+  const config: [string, IntlString, object][] = [
+    ['assigned', view.string.Assigned, {}],
```

Comment:
Added issue to adapt mode into URI: https://front.hc.engineering/workbench/platform/tracker/UBER-252